### PR TITLE
Scripts/ICC: Bosses now enter when in aggro range.

### PR DESCRIPTION
**Pull request by Golrag**
Closes https://github.com/TrinityCore/TrinityCore/issues/2134 :)

### CODE REVIEW

**Comment by Rushor** [comment]
the aggrorange is individual for every boss. some bosses start combat at 20 yards range, other bosses start combat at 5 yards range.

suggestion due to less correct information:

[Bossname] - [Range to Hitbox (3D)] - [Range to the middle of the creature (exact 3D)]
# Marrowgar - 4 - 25.5
# Lady Deathwhisper - 16 - 23.5
# Deathbringer Sauerfang - 20 - 23 
# Festergut - 15 - 24.5
# Rotface - 15 - 24.5
# Prof. Putricide - 5 - 14.4
# Bloodprince Council - 20 - 23.4

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp [hunk 1]
```diff
@@ -742,6 +730,33 @@ class spell_marrowgar_bone_slice : public SpellScriptLoader
...
+                        _introDone = true;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+    private:
+        bool _introDone;
```
Comment (Shauren) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp]: absolutely not.
There is only one at_lord_marrowgar_entrance loaded in memory shared by ALL instances - thats how all scripts work (single global instance, AI classes are slightly different, they are wrapped in script which creates a new object for each creature requesting it)

**Comment by Shauren** [comment]
@Rushor not quite. They simply follow regular aggro range calculations (with an upper cap) - being there at level 100 on retail significantly reduces aggro range

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp (hunk 1 above)
Comment (Golrag) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp]: Oh thanks for that information will move those back to the creatureAI ! :)

**Comment by Golrag** [comment]
@Shauren aggro range is quite high right now compared to https://www.youtube.com/watch?v=zFTceGrZIVk

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp [hunk 2]
```diff
@@ -742,6 +747,22 @@ class spell_marrowgar_bone_slice : public SpellScriptLoader
...
 };
 
+class at_lord_marrowgar_entrance : public AreaTriggerScript
+{
+    public:
+        at_lord_marrowgar_entrance() : AreaTriggerScript("at_lord_marrowgar_entrance") { }
+
+        bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/) override
+        {
+                if (InstanceScript* instance = player->GetInstanceScript())
```
Comment (MitchesD) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp]: tabs or weird codestyle? :D

Comment (Golrag) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp]: Forgot to backtab them when removing if (!_introDone), failed to notice this D:

**Comment by Golrag** [comment]
Should float Creature::GetAttackDistance(Unit const* player) const include ObjectSizes() ? looking at distances, it looks like it shouldn't. Can anyone confirm this ?

File: src/server/scripts/Northrend/IcecrownCitadel/boss_blood_prince_council.cpp [hunk 3]
```diff
@@ -1070,25 +1054,27 @@ class npc_blood_queen_lana_thel : public CreatureScript
...
-                if (Creature* summon = DoSummon(NPC_FLOATING_TRIGGER, triggerPos, 15000, TEMPSUMMON_TIMED_DESPAWN))
+                switch (action)
                 {
-                    summon->CastSpell(summon, SPELL_OOC_INVOCATION_VISUAL, true);
-                    summon->SetSpeed(MOVE_FLIGHT, 0.15f, true);
-                    summon->GetMotionMaster()->MovePoint(0, triggerEndPos);
+                    case ACTION_START_INTRO:
+                        if (!_introDone)
+                        {
+                            Talk(SAY_INTRO_1);
```
Comment (ccrs) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_blood_prince_council.cpp]: missing ``c++ _introDone = true;``

Comment (Golrag) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_blood_prince_council.cpp]: nice catch, thx :)

**Comment by Demonid** [comment]
When will this be merged?

**Comment by jackpoz** [comment]
what's the current status of this PR ?

**Comment by Golrag** [comment]
If what I stated above is true, then this PR is ready, core issue does not belong in this one :)

**Comment by Aokromes** [comment]
@Golrag i sugest you to jump into irc :)

**Pull request merged**